### PR TITLE
refactor with metaclass

### PR DESCRIPTION
**Pull request by 27149chen**
This pr use metaclass to implement the method of getting resource class by name, to replace the old way which heavily relies on the file name of the resource definition and is hard to maintain.
This pr also remove all the unnecessary __init__.py files, becasue python3 does not rely on them.

### CODE REVIEW

File: pyds8k/base.py [hunk 1]
```diff
@@ -72,7 +75,68 @@ def remove_None_fields_from_dict(self, input_dict):
...
 
 
+# all the resources are under folder "resources",
+# the route prefix of a resource resources/a/b/c.py is a.b
+def get_resource_route_prefix_by_class(cls):
+    path = os.path.abspath(
+        os.path.dirname(sys.modules[cls.__module__].__file__)
+    )
+    route = path.replace("/", ".")
+    route.rsplit(".resources.", 2)
```
Comment (shwzhang) [inline pyds8k/base.py]: this line may no need

File: pyds8k/base.py [hunk 2]
```diff
@@ -72,7 +75,68 @@ def remove_None_fields_from_dict(self, input_dict):
...
+    return route.rsplit(".resources.", 2)[1]
+
+
+class ResourceMeta(type):
+
+    def __new__(mcs, name, bases, dct):
+
+        new_class = super(ResourceMeta, mcs).__new__(mcs, name, bases, dct)
+        prefix = get_resource_route_prefix_by_class(new_class)
```
Comment (shwzhang) [inline pyds8k/base.py]: There is no need to get prefix if "resource_type" not in dict.

File: pyds8k/base.py [hunk 3]
```diff
@@ -72,7 +75,68 @@ def remove_None_fields_from_dict(self, input_dict):
...
+            RESOURCES[route] = new_class
+        return new_class
+
+
+class ManagerMeta(type):
+
+    def __new__(mcs, name, bases, dct):
+
+        new_class = super(ManagerMeta, mcs).__new__(mcs, name, bases, dct)
+        prefix = get_resource_route_prefix_by_class(new_class)
```
Comment (shwzhang) [inline pyds8k/base.py]: There is no need to get prefix if "resource_type" not in dict.

File: pyds8k/base.py (hunk 1 above)
Comment (27149chen) [reply pyds8k/base.py]: It is to support resources.cs.xxx.py

**Pull request merged**